filip/models/ngsi_v2/iot.py

```diff
@@ -292,18 +302,28 @@ class DeviceSettings(BaseModel):
         default=None,
         description="Optional Apikey key string to use instead of group apikey"
     )
-    endpoint: Optional[AnyHttpUrl] = Field(
+    endpoint: Optional[Union[AnyHttpUrl, str]] = Field(
         default=None,
         description="Endpoint where the device is going to receive commands, "
                     "if any."
     )
+    @field_validator('endpoint')
+    @classmethod
+    def validate_endpoint(cls, value):
```

Comment:
https://docs.pydantic.dev/latest/api/networks/

Reply:
I remove this code. As a consequence, `endpoint` will be a `Url` object if `model_dump` is called. The `Url` object can not be serialized to json automatically. However, if `model_dump_json` is called, it will be correctly converted to `str`